src/main/java/core/basesyntax/converter/DataConverterImpl.java

```diff
@@ -0,0 +1,50 @@
+package core.basesyntax.converter;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> data) {
+        List<FruitTransaction> convertedTransaction = new ArrayList<>();
+        for (int i = 1; i < data.size(); i++) { // Skip header
+            String[] parts = data.get(i).split(",");
+            if (parts.length != 3) {
+                System.err.println("Skipping invalid row (not enough data): " + data.get(i));
+                continue;
+            }
+
+            FruitTransaction.Operation operation;
+            try {
+                operation = mapOperation(parts[0]);
+            } catch (IllegalArgumentException e) {
+                System.err.println("Skipping invalid operation code in row: " + data.get(i));
+                continue;
+            }
+
+            String fruit = parts[1];
+            int quantity;
+            try {
+                quantity = Integer.parseInt(parts[2]);
+            } catch (NumberFormatException e) {
+                System.err.println("Skipping row with invalid quantity (not a valid number): "
+                        + data.get(i));
```

Comment:
Consider using a logging framework instead of `System.err.println` for better control over logging.